core-data/entities.js

```diff
@@ -34,4 +53,28 @@ export const getMethodName = ( kind, name ) => {
 	return `get${ kindPrefix }${ nameSuffix }`;
 };
 
-export default entities;
+/**
+ * Loads the kind entities into the store.
+ *
+ * @param {Object} state Global state
+ * @param {string} kind  Kind
+ *
+ * @return {Array} Entities
+ */
+export async function* getKindEntities( state, kind ) {
+	let entities = getEntitiesByKind( state, kind );
+
+	if ( entities && entities.length !== 0 ) {
+		return entities;
+	}
+
+	const kindConfig = find( kinds, { name: kind } );
+	if ( ! kindConfig ) {
+		return [];
+	}
+
+	entities = await kindConfig.loadEntities();
+	yield addEntities( entities );
+
+	return entities;
```

Comment:
The multipurposing of this function feels a bit awkward / forced. Not sure I have a great solution here. One option is to make it always yield the action, and extract this from the `getEntityRecord` resolver:

```js
export async function* getKindEntities( state, kind ) {
	const kindConfig = find( kinds, { name: kind } );
	if ( ! kindConfig ) {
		return;
	}

	let entities = getEntitiesByKind( state, kind );
	if ( ! entities ) {
		entities = await kindConfig.loadEntities();
	}

	yield addEntities( entities );
}
```

```js
export async function* getEntityRecord( state, kind, name, key ) {
	const { entities } = await ( await getKindEntities( state, kind ).next() ).value;
	const entity = find( entities, { kind, name } );
	if ( ! entity ) {
		return;
	}
	const record = await apiRequest( { path: `${ entity.baseUrl }/${ key }?context=edit` } );
	yield receiveEntityRecords( kind, name, record );
}
```

Makes `getEntityRecord` have strong awareness of what's yielded by `getKindEntities`, which could break down if ever `getKindEntities` yields more or different than just the action of `ADD_ENTITIES`. We could make `getKindEntities` not a generator, which could simplify it a bit more.

In retrospect, I might be more open to the idea of using `select` from within the resolver as was originally proposed, if it avoids some of the complexity here.

Reply:
I do agree, it's a bit weird, I'd like us to avoid adding complexity for the moment with a new API or something, let's try it for some time. We'll see how it plays when adding other entity selectors. I'm kind of hesitant to introduce `select` within a resolver just for the entities use-case.